ghost/admin/app/components/gh-migrate-iframe.js

```diff
@@ -0,0 +1,77 @@
+import Component from '@glimmer/component';
+import {action} from '@ember/object';
+// import {htmlSafe} from '@ember/template';
+import {inject as service} from '@ember/service';
+
+export default class GhMigrateIframe extends Component {
+    @service migrate;
+    @service router;
+    @service feature;
+    @service notifications;
+
+    willDestroy() {
+        super.willDestroy(...arguments);
+        window.removeEventListener('message', this.handleIframeMessage);
+    }
+
+    @action
+    setup() {
+        this.migrate.getMigrateIframe().src = this.migrate.getIframeURL();
+        window.addEventListener('message', this.handleIframeMessage);
+    }
+
+    @action
+    async handleIframeMessage(event) {
+        if (this.isDestroyed || this.isDestroying) {
+            return;
+        }
+
+        // only process messages coming from the migrate iframe
+        if (event?.data && this.migrate.getIframeURL().includes(event?.origin)) {
```

Comment:
Is it expected that a message can perform multiple actions? If not I would suggest returning after each one, or making them exclusive somehow to avoid any issues from running multiple!